src/app/state/organization/utils/get-main-orgs.ts

```diff
@@ -0,0 +1,40 @@
+export const getMainOrgs = ($getOrgList: any[]) => {
+  const list = [];
+  const subList = [];
+
+  // Separate Parent & Child Orgs
+  $getOrgList.forEach((item: any) => {
+    const namespaceArr = item.namespace.split('.');
+    if (namespaceArr.length === 3) {
+      list.push(item);
+    } else {
+      if (!subList[namespaceArr.length - 4]) {
+        subList[namespaceArr.length - 4] = [];
+      }
+      subList[namespaceArr.length - 4].push(item);
+    }
+  });
+
+  // Remove Unnecessary Sub-Orgs from Main List
+  if (list.length || subList.length) {
+    for (let i = 0; i < subList.length; i++) {
+      const arr = subList[i];
+      if (arr && arr.length) {
```

Comment:
The subsequent code doesn't seem to be referencing the `length` property of `arr`? Why the check for it in this `if`?

Reply:
Ok, this code is just extracted from governance-list.component.ts :D I didn't touched that. I just wrote few tests to be sure that it is working correctly. I will refactor this.
To let you know what is going on here, this code is getting main organization (for example test.iam.ewc) and all it's direct suborganization (suborg.test.iam.ewc etc.), but not sub-(...)-sub-org.